src/components/SwitzerlandEstimatedCasesDivisionModal.tsx

```diff
@@ -87,26 +88,23 @@ export const SwitzerlandEstimatedCasesDivisionModal = ({
   show,
   handleClose,
 }: Props) => {
-  const [caseCounts, setCaseCounts] = useState<CaseCountEntry[] | undefined>();
+  //const [caseCounts, setCaseCounts] = useState<CaseCountEntry[] | undefined>();
   const [showSwissRegions, setShowSwissRegions] = useState(true);
-
   const { dateFrom, dateTo, country } = variantSampleSet.sampleSelector;
-  useEffect(() => {
-    let isSubscribed = true;
-    const controller = new AbortController();
-    const signal = controller.signal;
 
-    getCaseCounts({ dateFrom, dateTo, country }, true, signal).then(caseCounts => {
-      if (isSubscribed) {
-        setCaseCounts(caseCounts);
-      }
-    });
+  const fetchCaseCounts = async () => {
+    const data = await getCaseCounts({ dateFrom, dateTo, country }, true, new AbortController().signal);
+    return data;
+  };
 
-    return () => {
-      isSubscribed = false;
-      controller.abort();
-    };
-  }, [dateFrom, dateTo, country]);
+  const { isLoading, isSuccess, error, isError, data: caseCounts, refetch } = useQuery<
+    CaseCountEntry[],
+    Error
+  >('caseCounts', fetchCaseCounts);
+
+  useEffect(() => {
+    refetch();
```

Comment:
We can call the `refetch` function which is returned from the `useQuery` in the useEffect hook whenever `dateFrom`, `dateTo` and `country` is updated.